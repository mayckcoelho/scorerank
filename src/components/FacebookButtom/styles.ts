import styled from 'styled-components/native';

export const StyledButtom = styled.TouchableOpacity`
    margin: 10px 0;
    border: 0;
    border-radius: 4px;
    padding: 16px;
    align-items: center;
    justify-content: center;
    background-color: #3b5998;

    flex-direction: row;
`

export const ButtomText = styled.Text`
    margin-left: 10px;
    font-weight: bold;
    color: #fff;
    font-size: 15px;
`