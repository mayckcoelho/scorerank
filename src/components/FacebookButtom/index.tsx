import React from 'react';

import { StyledButtom, ButtomText } from './styles';
import Icon from "react-native-vector-icons/FontAwesome";
Icon.loadFont();

const FacebookButtom: React.FC = () => (
    <StyledButtom>
        <Icon name="facebook" size={16} color="#fff" />
        <ButtomText>Continuar com Facebook</ButtomText>
    </StyledButtom>
);

export default FacebookButtom;