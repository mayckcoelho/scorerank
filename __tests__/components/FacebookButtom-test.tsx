import React from 'react';
import FacebookButtom from '../../src/components/FacebookButtom';
import renderer from 'react-test-renderer';

describe('FacebookButtom should render correctly', () => {
    it('match snapshot', () => {
        const facebookButtom = renderer
            .create(<FacebookButtom />)
            .toJSON();
        expect(facebookButtom).toMatchSnapshot();
    });
})