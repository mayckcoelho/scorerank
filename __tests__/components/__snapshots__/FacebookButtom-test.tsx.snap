// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookButtom should render correctly match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "borderColor": "black",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‚š
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginLeft": 10,
        },
      ]
    }
  >
    Continuar com Facebook
  </Text>
</View>
`;
